File: 2020/2020-06-26-lpc-dev-groot/_code/tree-r.cxx
auto f = TFile::Open("out.root", "READ");
auto t = f->Get<TTree>("t");

auto n = int32_t(0);
auto d = struct{
    int32_t i32;
	int64_t i64;
	double  f64;
};

t->SetBranchAddress("n", &n); // HL
t->SetBranchAddress("d", &d); // HL

for (int64_t i = 0; i < t->GetEntries(); i++) {
    t->GetEntry(i);
	printf("evt=%d, n=%d, d.i32=%d, d.i64=%d, d.f64=%f\n",
			i, n, d.i32, d.i64, d.f64);
}

File: 2020/2020-06-26-lpc-dev-groot/_code/tree-r.go
package main

import (
	"fmt"

	"go-hep.org/x/hep/groot"
	"go-hep.org/x/hep/groot/rtree"
)

func main() {
	// START OMIT
	f, err := groot.Open("f.root")
	defer f.Close()
	o, err := f.Get("t")

	v := struct {
		N int32 `groot:"n"`
		D struct {
			I32 int32   `groot:"i32"`
			I64 int64   `groot:"i64"`
			F64 float64 `groot:"f64"`
		} `groot:"d"`
	}{}

	r, err := rtree.NewReader(o.(rtree.Tree), rtree.ReadVarsFromStruct(&v))
	defer r.Close()

	err = r.Read(func(ctx rtree.RCtx) error { // HL
		fmt.Printf(
			"evt=%d, n=%d, d.i32=%d, d.i64=%d, d.f64=%v\n",
			ctx.Entry, v.N, v.D.I32, v.D.I64, v.D.F64,
		)
		return nil
	})
	// END OMIT
}
